app/src/main/res/layout/fragment_search.xml

```diff
@@ -15,34 +15,35 @@
   ~ limitations under the License.
   -->
 
-<RelativeLayout
+<FrameLayout
         xmlns:android="http://schemas.android.com/apk/res/android"
         android:layout_width="match_parent"
         android:layout_height="match_parent">
 
-    <ProgressBar
-            android:id="@+id/progressBar_searchProgress"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:visibility="gone"
-            android:indeterminate="true"
-            style="@android:style/Widget.ProgressBar.Horizontal"/>
 
     <android.support.v7.widget.RecyclerView
             android:id="@+id/recyclerView_searchResults"
             android:layout_width="match_parent"
             android:layout_height="match_parent"
-            android:layout_below="@+id/progressBar_searchProgress"
             android:scrollbars="vertical"/>
 
+    <!-- Negative margin because drawable has own unwanted padding -->
+    <ProgressBar
+            android:id="@+id/progressBar_searchProgress"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_gravity="top"
+            android:visibility="gone"
+            android:layout_marginTop="-4dp"
+            style="@style/ProgressBarStyle"/>
+
     <TextView
             android:id="@+id/textView_searchNoResults"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
-            android:layout_below="@+id/progressBar_searchProgress"
             android:layout_gravity="center"
```

Comment:
Maybe use the chance to change the gravity here, as one never sees that the query failed with the keyboard open. Another way would be to make the layout consider the keyboard as the bottom anchor if it's active (Not sure how easy this is though)